app/src/main/java/org/mozilla/focus/cfr/CfrMiddleware.kt

```diff
@@ -22,11 +25,39 @@ class CfrMiddleware(private val components: Components) : Middleware<BrowserStat
         action: BrowserAction
     ) {
         next(action)
-        if (action is TabListAction.AddTabAction) {
+
+        if (action is TabListAction.AddTabAction &&
+            Features.SHOW_ERASE_CFR &&
+            !components.appStore.state.showTrackingProtectionCfr
+        ) {
             components.settings.numberOfTabsOpened++
             if (components.settings.numberOfTabsOpened == ERASE_CFR_LIMIT) {
                 components.appStore.dispatch(AppAction.ShowEraseTabsCfrChange(true))
             }
         }
+
+        if (shouldShowCfrForTrackingProtection(action = action, context = context)) {
+            components.appStore.dispatch(AppAction.ShowTrackingProtectionCfrChange(true))
+        }
     }
+
+    // We shouldn't show CFR for Mozilla pages #6383.
+    private fun isMozillaUrl(context: MiddlewareContext<BrowserState, BrowserAction>) =
+        (context.state.findTabOrCustomTabOrSelectedTab(context.state.selectedTabId)?.content?.url)?.contains("mozilla")
```

Comment:
Since this change was not really needed for the tracking protection CFR refactoring it should've been part of a separate PR. It's recommended to avoid fixing more tickets in one PR, even more so in one commit. Easier in review, easier to track down changes later.

Reply:
Will be changed in a different pr